### DISCUSSION TITLE
Add new language to VITS pretrained model

### DISCUSSION THREAD

**Discussion by NeonBohdan** [coqui-ai/TTS#1496] [General Q&A] [answered]
[From here](https://coqui.ai/blog/tts/yourtts-zero-shot-text-synthesis-low-resource-languages#training) I can see that training happened with gradualy adding each of 3 languages in YourTTS (en, pt, fr)

But how to add new language to pretrained `tts_models/multilingual/multi-dataset/your_tts`
Will the `LanguageManager` automaticaly add a new language in the end of list in `language_ids.json`
If I do load YourTTS `config.json` and then add more datasets to it?

Or it is only possible by reinitializing `language_embedding` somehow?

**Comment by jeffersonquispe** [comment]
I have the same question

**Comment by erogol** [comment, marked as answer]
To add a new language, you need to stop training, update the config and restart training restoring the from the latest model checkpoint.

**Comment by NeonBohdan** [comment]
Thanks, now we can clone voices to implement more rare languages 
See [here](https://huggingface.co/spaces/neongeckocom/neon-tts-plugin-coqui)

**Reply by dostee-hemin** [reply]
Hi, could you explain how you went about training the YourTTS pre-trained model for new languages that it has never seen before? More specifically, what configurations did you change or add to the config.json file to accommodate for the new languages? Moreover, how did you write the train.py code to continue training from the pre-trained model using the new datasets? 

Any insight will be greatly appreciated.
